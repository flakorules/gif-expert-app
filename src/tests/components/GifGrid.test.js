import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme';

import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe('Pruebas en GifGrid', () => {

    const category = 'metallica';
    test('debe mostrarse correctamente ', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    })

    test('debe de mostrar items cuando se cargan imagenes', () => {

        const gifs = [{
            id: '666',
            url: 'https://www.666.cl/666.jpg',
            title: '666'
        },
        {
            id: '667',
            url: 'https://www.666.cl/666.jpg',
            title: '666'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={category} />);


        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);


    })





})
