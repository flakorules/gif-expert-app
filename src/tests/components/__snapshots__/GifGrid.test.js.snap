// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGrid debe de mostrar items cuando se cargan imagenes 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    metallica
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="666"
      key="666"
      title="666"
      url="https://www.666.cl/666.jpg"
    />
    <GifGridItem
      id="667"
      key="667"
      title="666"
      url="https://www.666.cl/666.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en GifGrid debe mostrarse correctamente  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    metallica
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
